customer-service/src/main/java/com/epam/customerservice/service/ProductService.java

```diff
@@ -0,0 +1,67 @@
+package com.epam.customerservice.service;
+
+import com.epam.customerservice.dto.ProductDto;
+import com.epam.customerservice.dto.SearchAndFilterRequestDto;
+import com.epam.customerservice.entity.ProductEntity;
+import com.epam.customerservice.helper.Indices;
+import com.epam.customerservice.repository.ElasticDatabaseRepository;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.search.SearchHit;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ProductService {
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    private RestHighLevelClient client;
+    private ElasticDatabaseRepository repository;
+    private ProductDto productDto;
+
+    public ProductService(RestHighLevelClient client, ElasticDatabaseRepository repository) {
+        this.client = client;
+        this.repository = repository;
+        productDto = new ProductDto();
+    }
+
+    public ProductDto findById(final Long id) {
+        return productDto.entityToDto(repository.findById(id).orElse(null));
+    }
+
+    public void save(final ProductDto productEntity) {
+        repository.save(productEntity.dtoToEntity());
+    }
+
+    public void deleteById(Long id) {
+        repository.deleteById(id);
+    }
+
+    public List<ProductDto> searchAndFilter(SearchAndFilterRequestDto requestDto) {
+        SearchRequest request = SearchAndFilterRequestBuilder.buildSearchAndFilterRequest(
+                Indices.PRODUCT_INDEX, requestDto);
+
+        SearchResponse response = null;
+        try {
+            response = client.search(request, RequestOptions.DEFAULT);
+            SearchHit[] searchHits = response.getHits().getHits();
+            List<ProductDto> products = new ArrayList<>(searchHits.length);
+
+            for(SearchHit hit : searchHits) {
+                products.add(objectMapper.readValue(hit.getSourceAsString(), ProductDto.class));
+            }
+            return products;
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
I'd throw an exception here without logging. Let the REST controller or the ExceptionHandler decide what to do with it.

Reply:
Fixed it